src/components/ProductOrder/ReceiverInfoSection.tsx

```diff
@@ -0,0 +1,55 @@
+import InputOrder from '@/common/InputOrder';
+import useOrderForm from '@/hooks/useOrderForm';
+import styled from '@emotion/styled';
+
+interface ReceiverInfoSectionProps {
+  receiverName: ReturnType<typeof useOrderForm>;
+  receiverPhone: ReturnType<typeof useOrderForm>;
+  quantity: ReturnType<typeof useOrderForm>;
```

Comment:
아래 코드를 살펴보니, `useOrderForm`이 `return` 하는 값들 중에서 `value`, `onChange`, `error` 만 있으면 되는것으로 보이는데요!
이럴때는 조금 더 명시적으로 필요한 값만 타입을 지정해주는게 필요한 데이터와 타입을 더 명확하게 알 수 있어 협업할 때 더욱 편하답니다!

```suggestion
  receiverName: Pick<
    ReturnType<typeof useOrderForm>,
    'error' | 'onChange' | 'value'
  >;
  receiverPhone: Pick<
    ReturnType<typeof useOrderForm>,
    'error' | 'onChange' | 'value'
  >;
  quantity: Pick<
    ReturnType<typeof useOrderForm>,
    'error' | 'onChange' | 'value'
  >;
```
<br/>
<br/>

추가로, 요렇게 특정 함수, 컴포넌트, 훅 등의 매게변수 혹은 리턴타입을 이용해서 타입을 지정하게 될 경우에는
타입을 사용하는쪽과 타입을 제공하는쪽이 강결합 되는데요 (`useOrderForm`과 `ReceiverInfoSection` 컴포넌트)

`ReceiverInfoSection` 컴포넌트가 Order 라는 도메인에 소속되어있는 도메인 컴포넌트이다보니, useOrderForm과 강결합되어도 크게 상관은 없고 오히려 관계를 표현해주어서 좋을 수 있지만,
후에 여러 도메인에서 사용할 범용적인 컴포넌트를 작성하신다면 강결합은 최대한 피해주시는게 좋을 것 같습니다!